src/applications/vaos/tests/new-appointment/components/CommunityCareProviderSelectionPage/index.unit.spec.jsx

```diff
@@ -302,6 +310,78 @@ describe('VAOS <CommunityCareProviderSelectionPage>', () => {
     );
   });
 
+  it('should display choose provider when remove provider clicked and address is null', async () => {
+    const testState = {
+      facilities: [{ facilityId: '983', isCerner: false }],
+      vapContactInfo: {
+        residentialAddress: null,
+        mailingAddress: {
+          addressLine1: '123 big sky st',
+          city: 'Cincinnati',
+          stateCode: 'OH',
+          zipCode: '45220',
+          latitude: 39.1,
+          longitude: -84.6,
+        },
+      },
+    };
+
+    const updateProfileFieldsExtra = payload => {
+      return {
+        type: 'UPDATE_USER_PROFILE_EXTRA',
+        payload,
+      };
+    };
+
+    const profile = (state = {}, action) => {
+      if (action.type === 'UPDATE_USER_PROFILE_EXTRA') {
+        return Object.assign({}, state, action.payload);
+      }
+
+      return state;
+    };
+
+    const store = createTestStoreWithUserProfile(initialState, profile);
+    await setTypeOfCare(store, /primary care/i);
+    await setTypeOfFacility(store, /Community Care/i);
+    const screen = renderWithStoreAndRouter(
+      <CommunityCareProviderSelectionPage />,
+      {
+        store,
+      },
+    );
+    // Choose Provider that is buried 2 clicks deep
+    userEvent.click(await screen.findByText(/Choose a provider/i));
+    userEvent.click(await screen.findByText(/more providers$/i));
+    userEvent.click(await screen.findByText(/more providers$/i));
+    userEvent.click(await screen.findByText(/AJADI, ADEDIWURA/i));
+    userEvent.click(
+      await screen.getByRole('button', { name: /choose provider/i }),
+    );
+    expect(screen.baseElement).to.contain.text(
+      'AJADI, ADEDIWURA700 CONSTITUTION AVE NEWASHINGTON, DC 20002-6599',
+    );
+
+    // set residential address to null - had to add a dispatch here because
+    // some of the test setup needed the address to exist for proper scaffolding
+    store.dispatch(updateProfileFieldsExtra(testState));
```

Comment:
No need for this dispatch

Reply:
@jenniemc  - I had to create this to null the address after the test steps had scaffolded the state. A null residentialAddress in initial state caused an exception in the other steps prior to this line.